runtime/src/main/java/org/corfudb/protocols/service/CorfuProtocolLogUnit.java

```diff
@@ -0,0 +1,539 @@
+     */
+    public static InspectAddressesResponse getInspectAddressesResponse(InspectAddressesResponseMsg msg) {
+        return new InspectAddressesResponse(msg.getEmptyAddressList());
+    }
+
+    /**
+     * Returns a TRIM_LOG (PREFIX_TRIM) request that can be sent by
+     * the client.
+     *
+     * @param address  an address to trim up to (i.e. [0, address))
+     * @return         a RequestPayloadMsg containing the TRIM_LOG request
+     */
+    public static RequestPayloadMsg getTrimLogRequestMsg(Token address) {
+        return RequestPayloadMsg.newBuilder()
+                .setTrimLogRequest(TrimLogRequestMsg.newBuilder()
+                        .setAddress(getTokenMsg(address))
+                        .build())
+                .build();
+    }
+
+    /**
+     * Returns a TRIM_LOG (PREFIX_TRIM) response that can be sent
+     * by the server.
+     *
+     * @return  a ResponsePayloadMsg containing the TRIM_LOG response
+     */
+    public static ResponsePayloadMsg getTrimLogResponseMsg() {
+        return ResponsePayloadMsg.newBuilder()
+                .setTrimLogResponse(TrimLogResponseMsg.getDefaultInstance())
+                .build();
+    }
+
+    /**
+     * Returns a TRIM_MARK request that can be sent by the client.
+     *
+     * @return  a RequestPayloadMsg containing the TRIM_MARK request
+     */
+    public static RequestPayloadMsg getTrimMarkRequestMsg() {
+        return RequestPayloadMsg.newBuilder()
+                .setTrimMarkRequest(TrimMarkRequestMsg.getDefaultInstance())
+                .build();
+    }
+
+    /**
+     * Returns a TRIM_MARK response that can be sent by the server.
+     *
+     * @param trimMark  the first untrimmed address in the address space
+     * @return          a ResponsePayloadMsg containing the TRIM_MARK response
+     */
+    public static ResponsePayloadMsg getTrimMarkResponseMsg(long trimMark) {
+        return ResponsePayloadMsg.newBuilder()
+                .setTrimMarkResponse(TrimMarkResponseMsg.newBuilder()
+                        .setTrimMark(trimMark)
+                        .build())
+                .build();
+    }
+
+    /**
+     * Returns a TAILS request of the specified type that can be sent by the client.
+     *
+     * @param type  the type of the TAILS request
+     * @return      a RequestPayloadMsg containing the TAILS request
+     */
+    public static RequestPayloadMsg getTailRequestMsg(TailRequestMsg.Type type) {
+        return RequestPayloadMsg.newBuilder()
+                .setTailRequest(TailRequestMsg.newBuilder()
+                        .setReqType(type)
+                        .build())
+                .build();
+    }
+
+    /**
+     * Returns a TAILS request of type STREAM_TAILS that can be sent by the client.
+     *
+     * @param streamIds  the list of stream IDs
+     * @return           a RequestPayloadMsg containing the TAILS request
+     */
+    public static RequestPayloadMsg getTailRequestMsg(List<UUID> streamIds) {
```

Comment:
Where is this method being used? Looks like only the one above this one, i.e. getTailRequestMsg with the type as parameter is being used. I think you can remove this one.

Reply:
It's only being used in some of the unit tests.

Reply:
Note: this has been [removed](https://github.com/CorfuDB/CorfuDB/pull/2836#discussion_r556715660).